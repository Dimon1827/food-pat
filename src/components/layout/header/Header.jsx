import { useLocation } from 'react-router-dom';
import Navigation from '../navigation/Navigation';
import Logo from './logo/Logo';
import Hero from './hero/Hero';
import './Header.scss';

function Header() {
  const location = useLocation();
  const isHome = location.pathname === '/';
  return (
    <header className ={`header ${isHome && 'header--container'}`}>
      <div className="container">
        <div className="header__wrapper header__wrapper--indent">
          <Logo />
          <Navigation />
        </div>
      </div>
      <Hero />
    </header>
  );
}

export default Header;
