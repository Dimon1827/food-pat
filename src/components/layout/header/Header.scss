@import '@/assets/styles/variables';

.header {
  position: relative;

  &--container {
    @media (min-width: $desktop-width) {
      margin-bottom: 82px;
    }
  }
}

.header.header--container {
  @media (min-width: $desktop-width) {
    background-image: linear-gradient(90deg, $white-color 50%, $bg-basic-orange 50%);
  }
}

.header__wrapper {
  padding-bottom: 3px;

  @media (min-width: $tablet-width) {
    display: flex;
    align-items: center;
    padding-top: 26px;
    margin-bottom: 65px;
  }

  @media (min-width: $desktop-width) {
    margin-bottom: 107px;
    padding-top: 58px;
    justify-content: space-between;
  }

  &--indent {
    @media (min-width: $tablet-width) {
      padding-bottom: 0;
    }

    @media (min-width: $desktop-width) {
      margin-bottom: 71px;
    }
  }
}

.header__logo {
  max-width: 253px;

  @media (min-width: $tablet-width) {
    margin-right: auto;
  }
}

.header__logo.logo {
  display: flex;
  align-items: center;

  &:hover {
    opacity: 0.8;
  }

  &:active {
    opacity: 0.6;
  }
}
