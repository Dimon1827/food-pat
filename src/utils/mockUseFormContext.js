import { useFormContext } from "react-hook-form";

export function mockUseFormContext(errors) {
  useFormContext.mockReturnValue({
    register: jest.fn(),
    formState: {
      errors: errors,
    },
  });
}


