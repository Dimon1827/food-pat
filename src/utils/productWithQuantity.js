const productWithQuantity = (food) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
