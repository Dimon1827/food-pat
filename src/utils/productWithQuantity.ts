import { IProducts } from "redux/slice/productsSlice.types";

interface IproductWithQuantity extends Omit<IProducts, 'isChecked' | 'quantity'> {};

const productWithQuantity = (food: IproductWithQuantity) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
