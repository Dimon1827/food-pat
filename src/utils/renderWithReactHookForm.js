import { render } from '@testing-library/react';
import { FormProvider, useForm } from 'react-hook-form';

export function renderWithReactHookForm(ui, { defaultValues = {} } = {}) {
  let reactHookFormMethods = {};

  const Wrapper = ({ children }) => {
    const methods = useForm({ defaultValues });
    return <FormProvider {...methods}>{children}</FormProvider>;
  };

  return {
    ...render(ui, { wrapper: Wrapper }),
  };
}