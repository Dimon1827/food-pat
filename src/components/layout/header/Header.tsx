import { useLocation } from 'react-router-dom';
import Navigation from '../navigation/Navigation';
import Logo from './logo/Logo';
import Hero from './hero/Hero';
import './Header.scss';

function Header() {
  const location = useLocation();
  const isHome = location.pathname === '/';
  return (
    <header className={`header ${isHome ? 'header--container' : ''}`}>
      <div className="container">
        <div className="header__wrapper header__wrapper--indent">
          <Logo className='header__logo'/>
          <Navigation isHome={isHome} />
        </div>
      </div>
      {!!isHome && <Hero />}
    </header>
  );
}

export default Header;
