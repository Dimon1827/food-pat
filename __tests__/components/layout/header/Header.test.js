import { screen } from '@testing-library/react';
import { renderWithRouter } from '@/utils/renderWithRouter';
import Header from '@/components/layout/header/Header';
import * as reduxHooks from 'react-redux';
import { useLocation } from 'react-router-dom';

jest.mock('react-redux');

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: jest.fn(),
}));


describe('Header', () => {

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Header renders correctly', () => {
    jest.spyOn(reduxHooks, 'useSelector').mockReturnValue(4);
    useLocation.mockReturnValue({ pathname: '/some-other-route' });
    renderWithRouter(<Header />);
    expect(screen.getByTestId('header-container')).toMatchSnapshot();
  });
  it('Header doesnt have class header--container if route in home', () => {
    jest.spyOn(reduxHooks, 'useSelector').mockReturnValue(4);
    useLocation.mockReturnValue({ pathname: '/' });
    renderWithRouter(<Header />);
    const headerContainer = screen.getByTestId('header-container');
    expect(headerContainer.classList.contains('header--container')).toBe(true);

  });
});
