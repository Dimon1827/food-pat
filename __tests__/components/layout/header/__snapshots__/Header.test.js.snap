// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header renders correctly 1`] = `
<header
  class="header "
  data-testid="header-container"
>
  <div
    class="container"
  >
    <div
      class="header__wrapper header__wrapper--indent"
    >
      <div
        class="logo header__logo"
      >
        <span
          class="main-logo"
          color="orange"
        />
      </div>
      <nav
        class="navigation navigation--closed"
      >
        <button
          class="navigation__toggle"
          type="button"
        >
          <span
            class="visually-hidden"
          >
            Открыть меню
          </span>
        </button>
        <ul
          class="navigation__list"
        >
          <li
            class="navigation__item "
          >
            <a
              aria-current="page"
              class="navigation__link navigation__link--new-color active"
              href="/"
            >
              Главная
            </a>
          </li>
          <li
            class="navigation__item "
          >
            <a
              class="navigation__link navigation__link--new-color"
              href="/catalog"
            >
              Каталог продукции
            </a>
          </li>
          <li
            class="navigation__item "
          >
            <a
              class="navigation__link navigation__link--new-color"
              href="/form"
            >
              Подбор программы
            </a>
          </li>
          <li
            class="navigation__item navigation__item--basket"
          >
            <a
              class="navigation__link navigation__link--new-color"
              href="/basket"
            >
              Корзина
            </a>
            <div
              class="navigation__badge"
            >
              <span
                class="MuiBadge-root css-1ka20jc-MuiBadge-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="ShoppingBasketIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m17.21 9-4.38-6.56c-.19-.28-.51-.42-.83-.42s-.64.14-.83.43L6.79 9H2c-.55 0-1 .45-1 1 0 .09.01.18.04.27l2.54 9.27c.23.84 1 1.46 1.92 1.46h13c.92 0 1.69-.62 1.93-1.46l2.54-9.27L23 10c0-.55-.45-1-1-1zM9 9l3-4.4L15 9zm3 8c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2"
                  />
                </svg>
                <span
                  class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular css-2ao5rv-MuiBadge-badge"
                >
                  4
                </span>
              </span>
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
